From 2dfc6133324e321e52f2bf7281bc2310f63ecc97 Mon Sep 17 00:00:00 2001
From: David Watson <dvd_anstow@hotmail.co.uk>
Date: Thu, 11 Nov 2021 12:56:38 +0000
Subject: [PATCH] Fix build break in cmake 3.22.0-rc2

The comparison fails in cmake 3.22.0-rc2 when LIB_SEARCH_PATH is empty. Adding quotes ensures there is always a string to compare to
---
 cmake/AWSSDKConfig.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/AWSSDKConfig.cmake b/cmake/AWSSDKConfig.cmake
index 83eac0b3b43..acd51317320 100644
--- a/cmake/AWSSDKConfig.cmake
+++ b/cmake/AWSSDKConfig.cmake
@@ -129,7 +129,7 @@ endif()
 get_filename_component(TEMP_PATH "${AWSSDK_CORE_LIB_FILE}" PATH)
 get_filename_component(TEMP_NAME "${TEMP_PATH}" NAME)
 
-while (NOT TEMP_NAME STREQUAL ${LIB_SEARCH_PREFIX})
+while (NOT TEMP_NAME STREQUAL "${LIB_SEARCH_PREFIX}")
     set(TEMP_PLATFORM_PREFIX "${TEMP_NAME}/${TEMP_PLATFORM_PREFIX}")
     get_filename_component(TEMP_PATH "${TEMP_PATH}" PATH)
     get_filename_component(TEMP_NAME "${TEMP_PATH}" NAME)
